pkg/controllers/node/emptiness.go

```diff
@@ -65,35 +62,17 @@ func (r *Emptiness) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisi
 		return reconcile.Result{}, nil
 	}
 
-	emptinessTimestamp, hasEmptinessTimestamp := n.Annotations[v1alpha5.EmptinessTimestampAnnotationKey]
-	if !empty {
-		if hasEmptinessTimestamp {
-			delete(n.Annotations, v1alpha5.EmptinessTimestampAnnotationKey)
-			logging.FromContext(ctx).Infof("Removed emptiness TTL from node")
-		}
-		return reconcile.Result{}, nil
-	}
-	// 3. Set TTL if not set
-	n.Annotations = lo.Assign(n.Annotations)
-	ttl := time.Duration(ptr.Int64Value(provisioner.Spec.TTLSecondsAfterEmpty)) * time.Second
-	if !hasEmptinessTimestamp {
+	_, hasEmptinessTimestamp := n.Annotations[v1alpha5.EmptinessTimestampAnnotationKey]
+	if !empty && hasEmptinessTimestamp {
+		delete(n.Annotations, v1alpha5.EmptinessTimestampAnnotationKey)
+		logging.FromContext(ctx).Infof("Removed emptiness TTL from node")
+	} else if empty && !hasEmptinessTimestamp {
+		n.Annotations = lo.Assign(n.Annotations)
```

Comment:
This line doesn't do anything now

Reply:
Pretty sure it panics if n.Annotations is nil. Check this. https://gist.github.com/chakrit/8157474#file-main-go-L15

Reply:
Ah, good call. Why not?

```go
n.Annotations = lo.Assign(n.Annotations, map[string]string{
  v1alpha5.EmptinessTimestampAnnotationKey: r.clock.Now().Format(time.RFC3339)
})
```

Reply:
Sure can do